controllers/cluster_controller.go

```diff
@@ -448,6 +483,24 @@ func (r *FoundationDBClusterReconciler) getStatusFromClusterOrDummyStatus(logger
 		}, nil
 	}
 
+	/*Get the status of all the pods in the cluster.
+	 *Until all the pods are up it will keep cluster in config missing state and ask reconciler to spawn required pods/pvc, etc.
+	 *Once all the pods are up and running it will force the cluster to reconcile
+	 */
+	podsStatus := r.getAllPodStatus(ctx, logger, cluster)
```

Comment:
What is the benefit of doing this check always if the if statement below is only executed during the initial reconciliation?

Reply:
@johscheuer From logs I've observed that this same flow is also used when a majority of pods are deleted and in the scenario I've described in PR description. From what I've seen if all the pods in the cluster are deleted multiple times without waiting for them to come up and running the reconciliation state and status has stale data. This stale state of cluster results in pods not coming up and cluster stuck in problem state.
If the state updates then cluster status shows the fields in if condition set to 1. I can try again to see if the other if conditions can be removed.

Reply:
Could you please create a reproducing example in our e2e test suite? Could you add some more information about the test you did and how the cluster was configured?